frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -144,6 +148,27 @@ local footerTextGeneratorMap = {
             end
         end
     end,
+    frontlight_warmth = function(footer)
+        local symbol_type = footer.settings.item_prefix
+        local prefix = symbol_prefix[symbol_type].frontlight_warmth
+        local powerd = Device:getPowerDevice()
+        if powerd:isFrontlightOn() then
+            if Device:isCervantes() or Device:isKobo() then
+                return (prefix .. " %d%%"):format(powerd:getWarmth())
```

Comment:
I was talking about that one ;).

This will return `nil` on Kobo/cervantes devices without amber leds ;). And that'll crash ;).

Reply:
i.e., I'd just leave a single branch, you can safely floor on every platform.

Something along the lines of

```lua
local warmth = powerd:getWarmth()
if warmth then
    return (prefix .. " %d%%"):format(math.floor(warmth))
end
```

Reply:
Actualy, you don't even need to floor *anywhere*.

printf("%d") does a cast, that truncates.

Reply:
> This will return `nil` on Kobo/Cervantes devices without amber leds ;). And that'll crash ;).

Ohh, yes!